File: src/main/java/com/warren/bt/BehaviorScript.java
package com.warren.bt;

import org.powbot.api.script.AbstractScript;

public abstract class BehaviorScript extends AbstractScript {
  private BehaviorTree behaviorTree;
  
  @Override
  public void onStart() {
    behaviorTree = behaviorTree();
  }

  @Override
  public final void poll() {
    behaviorTree.run();
  }
  
  public abstract BehaviorTree behaviorTree();
}

File: src/main/kotlin/com/warren/bt/BehaviorScript.kt
package com.warren.bt

import org.powbot.api.script.AbstractScript

abstract class BehaviorScript : AbstractScript() {
    private lateinit var behaviorTree: BehaviorTree

    override fun onStart() {
        try {
            behaviorTree = behaviorTree()
        } catch (e: Error) {
            logger.info(e.stackTraceToString())
//            controller.stop()
        }
    }

    override fun poll() {
        try {
            val s = behaviorTree.tick()
            if (s != Status.RUNNING) {
                behaviorTree.reset()
            }
        } catch (e: Error) {
            logger.info(e.message)
        }
    }


    abstract fun behaviorTree(): BehaviorTree
}
